engine/access/state_stream/backend/backend_executiondata.go

```diff
@@ -32,26 +34,70 @@ type ExecutionDataBackend struct {
 
 	subscriptionHandler  *subscription.SubscriptionHandler
 	executionDataTracker tracker.ExecutionDataTracker
+
+	executionResultProvider optimistic_sync.ExecutionResultInfoProvider
+	executionStateCache     optimistic_sync.ExecutionStateCache
 }
 
-func (b *ExecutionDataBackend) GetExecutionDataByBlockID(ctx context.Context, blockID flow.Identifier) (*execution_data.BlockExecutionData, error) {
-	header, err := b.headers.ByBlockID(blockID)
+// GetExecutionDataByBlockID retrieves execution data for a specific block by its block ID.
+//
+// CAUTION: this layer SIMPLIFIES the ERROR HANDLING convention
+//   - All errors returned are guaranteed to be benign. The node can continue normal operations after such errors.
+//   - To prevent delivering incorrect results to clients in case of an error, all other return values should be discarded.
+//
+// Expected errors:
+//   - access.DataNotFoundError - Returned when the requested data is not (yet) available.
+//     This includes:
+//   - Missing execution result info for the given block ID
+//   - Missing snapshot for the selected execution result
+//   - Missing block execution data
```

Comment:
You don't need to provide this much detail here. also, wrap types with `[]`
```suggestion
//   - [access.DataNotFoundError]: when data required to process the request is not available.
```